### PR TITLE
[WIP] Delete plugin directory

### LINKED ISSUE #57579: The plugin directory is not useful
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:

There is not much value in keeping a few components in plugin/cmd/ and plugin/pkg when everything else lives in cmd/ and pkg/

**What you expected to happen**:

Components that would be moved:
plugin/cmd/kube-scheduler
plugin/pkg/scheduler

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/cc @thockin 
/sig scheduling

### PR DESCRIPTION
**Pull request by misterikkit**
**What this PR does / why we need it**:

Simplify code organization

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #57579

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/OWNERS [hunk 1]
```diff
@@ -1,11 +1,13 @@
 reviewers:
   - brendandburns
+  - davidopp
   - dchen1107
   - lavalamp
   - smarterclayton
   - thockin
 approvers:
   - brendandburns
+  - davidopp
```
Comment (liggitt) [inline pkg/OWNERS]: New top level owner?

**Comment by yastij** [comment]
Also cc @resouer

**Comment by resouer** [comment]
@misterikkit Sorry I may miss the context, not sure if we already discussed this in sig or somewhere.

While for my understanding, components stayed in plugin may indicate that users are fine to customize it or implement their own version, e.g. customized scheduler. I'm aware users can always build the new binaries but it still concerns me a little bit by moving them all in `/pkg`.

Would like to know your thoughts :) or any shortcoming to keep the plugins pkg?  also cc @kubernetes/sig-scheduling-misc

**Comment by misterikkit** [comment]
@resouer You didn't miss any context. I created #57579, but this hasn't been discussed in any sigs. Your intuition is correct that moving these components into pkg does not prevent them from being customized/replaced.

Discussion probably belongs on the issue or in a mailing list, but just a couple points. Users customize/implements components in pkg & cmd as well as in plugin, and custom components don't get checked in to the plugin directory. In other words, I don't think moving things around makes them any less pluggable.

File: pkg/OWNERS (hunk 1 above)
Comment (misterikkit) [reply pkg/OWNERS]: Result of a somewhat naive merge. I can push this one into pkg/scheduler/OWNERS.

**Comment by k82cn** [comment]
No objection on removing `/plugin/`, as we prefer not to commit code of additional/customized plugins in upstream, so we don-t need to maintain this directory now :).

Maybe we need to get some feedback from sig-arch, @kubernetes/sig-architecture-pr-reviews  :).

**Comment by misterikkit** [comment]
/hold
I'm splitting this into multiple PRs.

**Comment by thockin** [comment]
This LGTM overall.  I appreciate the breakup :)

**Comment by timothysc** [comment]
Do you still need this one? I saw the other one merge already. 
@misterikkit

**Comment by misterikkit** [comment]
@timothysc  No, I plan to do this piece by piece.

**Pull request closed without merging**